src/Algebra/Consequences/Base.agda

```diff
@@ -28,6 +33,30 @@ module _ {ℓ} {f : Op₁ A} (_≈_ : Rel A ℓ) where
                                      Involutive _≈_ f
   reflexive∧selfInverse⇒involutive refl inv _ = inv refl
 
+module _ {ℓ} {_∙_ : Op₂ A} {0# 1# : A} (_≈_ : Rel A ℓ) where
+
+  private
+    _≉_ : Rel A ℓ
+    x ≉ y = ¬ (x ≈ y)
+
+  integral⇒noZeroDivisors : Integral _≈_ 1# 0# _∙_ → 1# ≉ 0# →
+                            NoZeroDivisors _≈_ 0# _∙_
+  integral⇒noZeroDivisors (inj₁ trivial)        = contradiction trivial
+  integral⇒noZeroDivisors (inj₂ noZeroDivisors) = λ _ → noZeroDivisors
+
+module _ {ℓ} {_∙_ : Op₂ A} {0# : A} (_≈_ : Rel A ℓ) where
+
+  private
+    _≉_ : Rel A ℓ
+    x ≉ y = ¬ (x ≈ y)
+
+  noZeroDivisors⇒x≉0∧y≉0⇒xẏ≉0 : NoZeroDivisors _≈_ 0# _∙_ →
+                                x ≉ 0# → y ≉ 0# → (x ∙ y) ≉ 0#
+  noZeroDivisors⇒x≉0∧y≉0⇒xẏ≉0 noZeroDivisors x≉0 y≉0 xy≈0 with noZeroDivisors xy≈0
```

Comment:
don't have time to fully try, but could this be
```agda
either x≉0 y≉0 (noZeroDivisors xy≈0)
```

Reply:
Ah!? Will investigate tomorrow...

Reply:
The eliminator for disjoint union.